src/box/relay.cc

```diff
@@ -471,7 +471,9 @@ relay_initial_join(struct iostream *io, uint64_t sync, struct vclock *vclock,
 	 * Start sending data only when the latest sync
 	 * transaction is confirmed.
 	 */
-	if (txn_limbo_wait_confirm(&txn_limbo) != 0)
+	double timeout = replication_synchro_timeout_enabled ?
+		replication_synchro_timeout : replication_disconnect_timeout();
```

Comment:
Nit: let's use the same timeouts in all such places. Either `raft->death_timeout` or `replication_disconnect_timeout()`. I suggest the latter one.